File: spring-angular/roma/v1/backend-ms1/src/main/java/ma/zs/stocky/service/impl/admin/commun/ClientAdminServiceImpl.java
package ma.zs.stocky.service.impl.admin.commun;


import ma.zs.stocky.bean.core.commun.Client;
import ma.zs.stocky.dao.criteria.core.commun.ClientCriteria;
import ma.zs.stocky.dao.facade.core.commun.ClientDao;
import ma.zs.stocky.dao.specification.core.commun.ClientSpecification;
import ma.zs.stocky.service.facade.admin.commun.ClientAdminService;
import ma.zs.stocky.zynerator.service.AbstractServiceImpl;
import ma.zs.stocky.zynerator.util.ListUtil;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.ArrayList;





import org.springframework.beans.factory.annotation.Autowired;


import java.util.List;
@Service
public class ClientAdminServiceImpl extends AbstractServiceImpl<Client, ClientCriteria, ClientDao> implements ClientAdminService {






    public Client findByReferenceEntity(Client t){
        return  dao.findByCin(t.getCin());
    }


    public List<Client> findAllOptimized() {
        return dao.findAllOptimized();
    }





    public void configure() {
        super.configure(Client.class, ClientSpecification.class);
    }



    public ClientAdminServiceImpl(ClientDao dao) {
        super(dao);
    }

}

File: nest/v1/backend-ms1-new/src/app/module/service/admin/commun/ClientAdminServiceImpl.ts
import {Injectable, NotFoundException} from "@nestjs/common";

import {AbstractServiceImpl} from "src/app/zynerator/service/AbstractServiceImpl";

import {ClientDao} from "src/app/controller/dao/facade/core/commun/ClientDao";
import {ClientCriteria} from "src/app/controller/dao/criteria/core/commun/ClientCriteria";
import {Client} from "src/app/controller/bean/core/commun/Client";
import {ClientDto} from "src/app/controller/dto/commun/ClientDto";


@Injectable()
export class ClientAdminServiceImpl extends AbstractServiceImpl<Client, ClientCriteria, ClientDao>{

    constructor(private readonly dao: ClientDao ,
    ) {
        super(dao);
    }

    async save(item: Client): Promise<Client> {
        const saved = await this.dao.save(item);
        return saved;
    }


    async update(item: Client): Promise<Client> {
        const saved = await this.dao.saveOrUpdate(item);
        return saved;
    }

    async updateMultiple(items: Client[]): Promise<void> {
        if (items) {
            items.forEach(e => this.update(e))
        }
    }

    async  findAllOptimized(): Promise<ClientDto[]> {
        return this.dao.findAllOptimized();
    }

    async findAll(): Promise<Client[]> {
        return this.dao.findAll();
    }

    async findById(id: number): Promise<Client> {
        return this.dao.findById(id);
    }


    async deleteById(id: number): Promise<number> {
        const existing = await this.findById(id);
        if (!existing) {
            throw new NotFoundException(`Client with ID ${id} not found.`);
        }
        const result = await this.dao.deleteById(existing.id);
        return result;
    }

    async deleteMultiple(items: Client[]): Promise<Client[]> {
        const deletedItems: Client[] = [];
        for (const item of items) {
            await this.deleteById(item.id);
            deletedItems.push(item);
        }
        return deletedItems;
    }


    async findWithAssociatedLists(id: number): Promise<Client> {
        const result = await this.dao.findById(id);
    return result;
    }

    async updateWithAssociatedLists(item: Client): Promise<Client> {
         return await this.dao.saveOrUpdate(item);
    }
}

